src/ReverseDiff.jl

```diff
@@ -24,7 +24,7 @@ using ChainRulesCore
 # Not all operations will be valid over all of these types, but that's okay; such cases
 # will simply error when they hit the original operation in the overloaded definition.
 const ARRAY_TYPES = (:AbstractArray, :AbstractVector, :AbstractMatrix, :Array, :Vector, :Matrix)
-const REAL_TYPES = (:Bool, :Integer, :(Irrational{:e}), :(Irrational{:π}), :Rational, :BigFloat, :BigInt, :AbstractFloat, :Real, :Dual)
+const REAL_TYPES = (:Bool, :Integer, :(Irrational{:ℯ}), :(Irrational{:π}), :Rational, :BigFloat, :BigInt, :AbstractFloat, :Real, :Dual)
```

Comment:
This was a bug it seems:
```julia
julia> typeof(ℯ)
Irrational{:ℯ}

julia> typeof(MathConstants.e)
Irrational{:ℯ}
```

Reply:
Was it never tested? Would be nice to add a test.

Reply:
I already added a test (I only noticed the bug since it failed without the fix).